GDJS/Runtime/CustomRuntimeObject.ts

```diff
@@ -0,0 +1,587 @@
+      this._scaleX = newScale * (this._flippedX ? -1 : 1);
+      this.invalidateHitboxes();
+      this.getRenderer().update();
+    }
+
+    /**
+     * Change the scale on Y axis of the object (changing its height).
+     *
+     * @param newScale The new scale (must be greater than 0).
+     */
+    setScaleY(newScale: number): void {
+      if (newScale < 0) {
+        newScale = 0;
+      }
+      if (newScale === Math.abs(this._scaleY)) {
+        return;
+      }
+      this._scaleY = newScale * (this._flippedY ? -1 : 1);
+      this.invalidateHitboxes();
+      this.getRenderer().update();
+    }
+
+    /**
+     * Get the scale of the object (or the average of the X and Y scale in case
+     * they are different).
+     *
+     * @return the scale of the object (or the average of the X and Y scale in
+     * case they are different).
+     */
+    getScale(): number {
+      return (Math.abs(this._scaleX) + Math.abs(this._scaleY)) / 2.0;
+    }
+
+    /**
+     * Get the scale of the object on Y axis.
+     *
+     * @return the scale of the object on Y axis
+     */
+    getScaleY(): float {
+      return Math.abs(this._scaleY);
+    }
+
+    /**
+     * Get the scale of the object on X axis.
+     *
+     * @return the scale of the object on X axis
+     */
+    getScaleX(): float {
+      return Math.abs(this._scaleX);
+    }
+
+    // Visibility and display :
+    /**
+     * Change the transparency of the object.
+     * @param opacity The new opacity, between 0 (transparent) and 255 (opaque).
+     */
+    setOpacity(opacity: float): void {
+      if (opacity < 0) {
+        opacity = 0;
+      }
+      if (opacity > 255) {
+        opacity = 255;
+      }
+      this.opacity = opacity;
+      this.getRenderer().updateOpacity();
+    }
+
+    /**
+     * Get the transparency of the object.
+     * @return The opacity, between 0 (transparent) and 255 (opaque).
+     */
+    getOpacity(): number {
+      return this.opacity;
+    }
+
+    /**
+     * Hide (or show) the object
+     * @param enable true to hide the object, false to show it again.
+     */
+    hide(enable: boolean): void {
+      if (enable === undefined) {
+        enable = true;
+      }
+      this.hidden = enable;
+      this.getRenderer().updateVisibility();
+    }
+
+    flipX(enable: boolean) {
+      if (enable !== this._flippedX) {
+        this._scaleX *= -1;
+        this._flippedX = enable;
+        this.invalidateHitboxes();
+        this.getRenderer().update();
+      }
+    }
+
+    flipY(enable: boolean) {
+      if (enable !== this._flippedY) {
+        this._scaleY *= -1;
+        this._flippedY = enable;
+        this.invalidateHitboxes();
+        this.getRenderer().update();
+      }
+    }
+
+    isFlippedX(): boolean {
+      return this._flippedX;
+    }
+
+    isFlippedY(): boolean {
+      return this._flippedY;
+    }
+  }
+
+  // Others initialization and internal state management :
+  CustomRuntimeObject.supportsReinitialization = true;
```

Comment:
Is it really useful? Most objects seems to set it to false.

Reply:
In theory it could be a nice time/performance saver if you for example shoot a lot of bullets and destroy them, it would avoid creating/destroying objects and just "reset" a bunch of properties. But I don't have real benchmarks to prove this unfortunately.

Reply:
ok, it's probably something that can be done in a second version to be safe?

Reply:
I guess so yes. You can disable this for now